Zend/zend_builtin_functions.c

```diff
@@ -1061,6 +1067,29 @@ ZEND_FUNCTION(trait_exists)
 }
 /* }}} */
 
+ZEND_FUNCTION(is_enum)
+{
+	zval *name_or_object;
+	zend_string *name;
+	zend_bool autoload = 1;
+
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_ZVAL(name_or_object)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_BOOL(autoload)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (Z_TYPE_P(name_or_object) == IS_OBJECT) {
+		name = Z_OBJCE_P(name_or_object)->name;
```

Comment:
It's really inefficient to do a hash table lookup for the class entry by name when you already have the class entry. class_exists was never called with an object - http://php.net/class_exists

The following should work and be much faster:

```suggestion
		RETURN_BOOL(Z_OBJCE_P(name_or_object)->ce_flags & ZEND_ACC_ENUM);
```

Reply:
`is_enum` was replaced with `enum_exists` which only accepts strings. Thus this is obsolete.